jinahub/rankers/CatBoostRanker/catboost.py

```diff
@@ -0,0 +1,95 @@
+from typing import Dict
+
+import numpy as np
+from catboost import CatBoostRanker, Pool
+
+from jina import DocumentArray, Executor, requests
+from jina_commons.batching import get_docs_batch_generator
+
+
+class CatBoostRanker(Executor):
+    def __init__(
+        self,
+        query_features: List[str],
+        document_features: List[str],
+        label: str,
+        model_path: str = None,
+        catboost_parameters: Dict = {
+            'iterations': 2000,
+            'custom_metric': ['NDCG', 'PFound', 'AverageGain:top=10'],
+            'verbose': False,
+            'random_seed': 0,
+            'loss_function': 'RMSE',
+        },
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.q_features = query_features
+        self.d_features = document_features
+        self.label = label
+        self.model_path = model_path
+        self.catboost_parameters = catboost_parameters
+        self.model = None
+        if self.model_path and os.path.exists(self.model_path):
```

Comment:
u can call `self.load` directly to enforce the design that restarting a Pea is like reloading the model